docs/frameworks/sveltekit.md

```diff
@@ -0,0 +1,219 @@
+
+<details>
+<summary>1) add <strong>pwa.js</strong> script</summary>
+
+```js
+import { resolveConfig } from 'vite'
+import { VitePWA } from 'vite-plugin-pwa';
+import { pwaConfiguration } from './pwa-configuration.js';
+import { copyFileSync } from 'fs';
+
+const webmanifestDestinations = [
+	'./.svelte-kit/output/client/',
+	'./build/',
+]
+
+const swDestinations = [
+	'./build/',
+]
+
+const buildPwa = async() => {
+	const config = await resolveConfig({ plugins: [VitePWA({ ...pwaConfiguration })] }, 'build', 'production' )
+	// when `vite-plugin-pwa` is present, use it to regenerate SW after rendering
+	const pwaPlugin = config.plugins.find(i => i.name === 'vite-plugin-pwa')?.api
+	if (pwaPlugin?.generateSW) {
+		console.log('Generating PWA...')
+		await pwaPlugin.generateSW()
+		webmanifestDestinations.forEach(d => {
+			copyFileSync('./.svelte-kit/output/client/_app/manifest.webmanifest', `${d}/manifest.webmanifest`)
+		})
+		// don't copy workbox, SvelteKit will copy it
+		swDestinations.forEach(d => {
+			copyFileSync('./.svelte-kit/output/client/sw.js', `${d}/sw.js`)
+		})
+		console.log('Generation of PWA complete')
+	}
+} 
+
+buildPwa()
+```
+</details>
+
+
+<details>
+<summary>2) add <strong>pwa-configuration.js</strong> script</summary>
+
+```js
+const pwaConfiguration = {
+	srcDir: './build',
+	outDir: './.svelte-kit/output/client',
+	includeManifestIcons: false,
+	base: '/',
+	scope: '/',
+	manifest: {
+	short_name: "<YOUR APP SHORT NAME>",
+	name: "<YOUR APP NAME>",
+	scope: "/",
+	start_url: "/",
+	display: "standalone",
+	theme_color: "#ffffff",
+	background_color: "#ffffff",
+	icons: [
+		{
+			src: "/pwa-192x192.png",
+			sizes: "192x192",
+			type: "image/png"
+		},
+		{
+			src: "/pwa-512x512.png",
+			"sizes": "512x512",
+			"type": "image/png"
+		},
+		{
+			src: "/pwa-512x512.png",
+			"sizes": "512x512",
+			"type": "image/png",
+			purpose: 'any maskable'
+		}
+	]
+	},
+	workbox: {
+		// mode: 'development',
+		navigateFallback: '/',
+		// vite and SvelteKit are not aligned: pwa plugin will use /\.[a-f0-9]{8}\./ by default: #164 optimize workbox work
+		dontCacheBustURLsMatching: /-[a-f0-9]{8}\./,
+		globDirectory: './build/',
+		globPatterns: ['robots.txt', '**/*.{js,css,html,ico,png,svg,webmanifest}'],
+		globIgnores: ['**/sw*', '**/workbox-*'],
+		manifestTransforms: [async(entries) => {
+			// manifest.webmanifest is added always by pwa plugin, so we remove it.
+			// EXCLUDE from the sw precache sw and workbox-*
+			const manifest = entries.filter(({ url }) =>
+				url !== 'manifest.webmanifest' && url !== 'sw.js' && !url.startsWith('workbox-')
+			).map((e) => {
+				let url = e.url;
+				if (url && url.endsWith('.html')) {
+					if (url.startsWith('/'))
+						url = url.slice(1)
+					
+					e.url = url === 'index.html' ? '/' : `/${url.substring(0, url.lastIndexOf('/'))}`
```

Comment:
let's future-proof this so that it works even after https://github.com/sveltejs/kit/issues/1443
```suggestion
					if (url.startsWith('/')) {
						url = url.slice(1)
					}
					if (url === 'index.html') {
						e.url = url === 'index.html' ? '/'
					} else if (url.endsWith('index.html')) {
						e.url = `/${url.substring(0, url.lastIndexOf('/'))}`
					} else if (url.endsWith('.html')) {
						e.url = `/${url.substring(0, url.length - '.html'.length)}`
					}
```